rand_distr/src/poisson.rs

```diff
@@ -137,29 +137,41 @@ mod test {
     fn test_poisson_10() {
         let poisson = Poisson::new(10.0).unwrap();
         let mut rng = crate::test::rng(123);
-        let mut sum = 0;
+        let mut sum_u64 = 0;
+        let mut sum_f64 = 0.;
```

Comment:
What is the goal of this test? We know both values should be the same since (a) both should be non-negative integers and (b) we should not be going anywhere close to the limits/accuracy of either type.

Reply:
The goal is to test that the code path works as expected, not the precision.

Reply:
It's the same code path, aside from the extra cast.

Reply:
I would like to trigger the path with the cast, making sure it works. I think it is important because of the potential undefined behavior.